Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -515,13 +516,40 @@ internal bool AddSession(SFSession session, bool ensureMinPoolSize)
             }
         }
 
+        internal void DestroyPool()
+        {
+            s_logger.Debug("SessionPool::DestroyPool" + PoolIdentification());
+            lock (_sessionPoolLock)
+            {
+                _underDestruction = true;
+                ClearIdleSessions();
```

Comment:
is this still clearing just idle sessions whereas below DestroyPoolAsync calls ClearAllPoolsAsync?

Reply:
yes, I renamed ClearAllPoolsAsync to ClearIdleSessionsAsync()